### PR TITLE
Check that session and env match

### PR DESCRIPTION
**Pull request by alexandraBara**
Added check for fin_builder/eval to ensure that session assumed rocm_v and miopen_v  match the runtime rocm/miopen_v

### CODE REVIEW

File: tuna/fin_builder.py [hunk 1]
```diff
@@ -91,6 +91,13 @@ def process_pdb_compile(self, session, fin_json):
 
   def step(self):
     """Main functionality of the builder class. It picks up jobs in new state and compiles them"""
+
+    try:
+      self.check_env()
+    except ValueError as verr:
+      self.logger.error(verr)
+      return False
```
Comment (cderb) [inline tuna/fin_builder.py:99]: This will add extra file I/O for each job. Is there a clean way to check once before we launch the worker?

**Review by cderb — CHANGES_REQUESTED** [review]

**Review by reidkwja — APPROVED** [review]
Looks good!

File: tuna/fin_builder.py (hunk 1 above)
Comment (alexandraBara) [reply tuna/fin_builder.py:99]: ill investigate moving this call somewhere else

File: tuna/fin_eval.py [hunk 2]
```diff
@@ -213,6 +213,12 @@ def update_fdb_eval_entry(self, session, fdb_obj):
 
   def process_fdb_eval(self, fin_json, result_str='miopen_find_eval_result'):
     """process find db eval json results"""
+    try:
+      self.check_env()
+    except ValueError as verr:
+      self.logger.error(verr)
+      return False
```
[the remark this answers is not in the document]
Comment (alexandraBara) [reply tuna/fin_eval.py]: this was a mistake, i moved it right at the top of step

File: tuna/fin_builder.py (hunk 1 above)
Comment (cderb) [reply tuna/fin_builder.py:100]: Perhaps in the constructer for the Fin class objects? Or we could store state here to only check once during a run?

**Review by cderb — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Pull request merged**